pkg/postgres/migrations/202411131200_eigenStateModelConstraints/up.go

```diff
@@ -0,0 +1,32 @@
+package _202411131200_eigenStateModelConstraints
+
+import (
+	"database/sql"
+	"gorm.io/gorm"
+)
+
+type Migration struct {
+}
+
+func (m *Migration) Up(db *sql.DB, grm *gorm.DB) error {
+	queries := []string{
+		`alter table disabled_distribution_roots add constraint uniq_disabled_distribution_root unique (transaction_hash, log_index, block_number)`,
+		`alter table operator_share_deltas add constraint uniq_operator_share unique (transaction_hash, log_index, block_number, operator, staker, strategy)`,
+		`alter table reward_submissions add constraint uniq_reward_submission unique (transaction_hash, log_index, block_number, reward_hash, strategy_index)`,
+		`alter table reward_submissions drop constraint reward_submissions_reward_hash_strategy_block_number_key`,
+		`alter table staker_delegation_changes add constraint uniq_staker_delegation_change unique (transaction_hash, log_index, block_number)`,
+		`alter table staker_share_deltas add constraint uniq_staker_share_delta unique (transaction_hash, log_index, block_number, staker, strategy, strategy_index)`,
+		`alter table submitted_distribution_roots add constraint uniq_submitted_distribution_root unique (transaction_hash, log_index, block_number)`,
```

Comment:
You actually don't need `block_number` here. 

If it's 1:1 to the log, you just need `transaction_hash` and `log_index`.

Reply:
For all other models that's not 1:1 to the logs, you just need one or two additional items to distinguish the rows.

Reply:
i.e just `strategyIndex` instead of `strategy` as well.

Reply:
I'm pretty sure there's a performance hit to doing a compound unique constraint (i.e fewer fields means better performance). So we should strive to have just the necessary fields.

Reply:
> there's a performance hit to doing a compound unique constraint

On insert there is, but thats basically it, and we're performing (relatively speaking) few inserts on a per-block basis that it's not going to make a noticeable difference. Thats my assumption at least, I'll profile it and see how it goes.